src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php

```diff
@@ -48,13 +48,16 @@ public function createListFromLoader(ChoiceLoaderInterface $loader, $value = nul
     public function createView(ChoiceListInterface $list, $preferredChoices = null, $label = null, $index = null, $groupBy = null, $attr = null)
     {
         $preferredViews = [];
+        $preferredViewsOrder = [];
         $otherViews = [];
         $choices = $list->getChoices();
         $keys = $list->getOriginalKeys();
 
         if (!\is_callable($preferredChoices) && !empty($preferredChoices)) {
-            $preferredChoices = function ($choice) use ($preferredChoices) {
-                return \in_array($choice, $preferredChoices, true);
+            // make sure we have keys that reflect order
```

Comment:
I don't understand the end of this comment. It'd understand it better this way, but I don't know if it's precise enough: `.. that reflect order` -> `... that keep the preferred order` ?

Reply:
Do you like this more?
```suggestion
            // make sure we have keys that represent order
```

Reply:
Let's keep the original code comment and forget my comment :)